unit tests, volumes: Refactor in order to reduce unrequired parameters and ease readability

oshoval opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

**What this PR does / why we need it**:
Discard function boolean arg by splitting to two functions.
Discard unrequired args which are the same all over the table entries.

Split DV and PVC to two DescribeTable blocks.
It ease maintenance, when each blocks is focused on one time,
and removes args and condition.
The common code that it saved is not big and added mental burden.

Inline single usage functions, and get rid of the offset.

Replaced strings with consts.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
None
```

oshoval commented:
/sig code-quality

0xFelix commented on pkg/virtctl/vm/add_volume_test.go:
```diff
@@ -133,42 +135,53 @@ var _ = Describe("Add volume command", func() {
-		expectFunc(vmiName, volumeName, useDv)
-		commandAndArgs := append([]string{commandName, vmiName, fmt.Sprintf("--volume-name=%s", volumeName)}, args...)
+	DescribeTable("with DataVolume addvolume cmd, should call correct endpoint", func(expectFunc func(vmiName, volumeName string, verifyFunc VerifyFunc), args ...string) {
+		vmiName := "testvmi"
+		volumeName := "testvolume"
```
Make these `const`?

0xFelix commented on pkg/virtctl/vm/add_volume_test.go:
```diff
@@ -133,42 +135,53 @@ var _ = Describe("Add volume command", func() {
+	)
+
+	DescribeTable("with PVC addvolume cmd, should call correct endpoint", func(expectFunc func(vmiName, volumeName string, verifyFunc VerifyFunc), args ...string) {
+		vmiName := "testvmi"
+		volumeName := "testvolume"
```
Same

0xFelix commented on pkg/virtctl/vm/add_volume_test.go:
```diff
@@ -180,13 +193,14 @@ func createTestPVC() *k8sv1.PersistentVolumeClaim {
-		ExpectWithOffset(3, volumeOptions.VolumeSource.PersistentVolumeClaim).To(BeNil())
-	} else {
-		ExpectWithOffset(3, volumeOptions.VolumeSource.DataVolume).To(BeNil())
-		ExpectWithOffset(3, volumeOptions.VolumeSource.PersistentVolumeClaim).ToNot(BeNil())
-	}
+	ExpectWithOffset(3, volumeOptions.VolumeSource.DataVolume).ToNot(BeNil())
+	ExpectWithOffset(3, volumeOptions.VolumeSource.PersistentVolumeClaim).To(BeNil())
+}
+
+func verifyPVCVolumeSource(volumeOptions *v1.AddVolumeOptions) {
+	Expect(volumeOptions.VolumeSource).ToNot(BeNil())
+	ExpectWithOffset(3, volumeOptions.VolumeSource.DataVolume).To(BeNil())
+	ExpectWithOffset(3, volumeOptions.VolumeSource.PersistentVolumeClaim).ToNot(BeNil())
 }
```
WDYT about inlining these, as they are both used only a single time?

oshoval replied:
```
		expectFunc(vmiName,
			volumeName,
			func(volumeOptions *v1.AddVolumeOptions) {
				Expect(volumeOptions.VolumeSource).ToNot(BeNil())
				ExpectWithOffset(3, volumeOptions.VolumeSource.DataVolume).ToNot(BeNil())
				ExpectWithOffset(3, volumeOptions.VolumeSource.PersistentVolumeClaim).To(BeNil())
			})
```
Imo it isn't good because it force the reader to go into details while reading high level

oshoval commented:
Addressed comments, thanks (answered inline otherwise)

oshoval commented:
Addressed comments:
Inline functions that are used one time
Note that inlining them also allows to use Expect without offset
which is easier to maintain and shows the right stack line.

oshoval commented:
Replaced some hardcoded strings with const usage
it is also more consistent of course

0xFelix commented on pkg/virtctl/vm/add_volume_test.go:
```diff
@@ -133,60 +136,76 @@ var _ = Describe("Add volume command", func() {
-					k8smetav1.CreateOptions{})
-			}
+	DescribeTable("with DataVolume addvolume cmd, should call correct endpoint", func(expectFunc func(vmiName, volumeName string, verifyFunc VerifyFunc), args ...string) {
+		const (
+			vmiName    = "testvmi"
+			volumeName = "testvolume"
+		)
```
nit: Dedup consts instead of redefining them in the test blocks.

oshoval commented:
Deduped consts and refactor more strings repetitions

oshoval commented:
Unexport verifyFunc

0xFelix reviewed (approved):
/lgtm

enp0s3 reviewed (commented):
/approve
@oshoval Thank you for the effort!

oshoval commented:
Thanks

/retest

oshoval commented:
/test pull-kubevirt-client-python
